tokio/src/sync/batch_semaphore.rs

```diff
@@ -150,12 +185,25 @@ impl Semaphore {
         // currently we just clamp the permit count when it exceeds the max
         permits &= Self::MAX_PERMITS;
 
-        Self {
-            permits: AtomicUsize::new(permits << Self::PERMIT_SHIFT),
-            waiters: Mutex::const_new(Waitlist {
-                queue: LinkedList::new(),
-                closed: false,
-            }),
+        cfg_trace! {
+            return Self {
+                permits: AtomicUsize::new(permits << Self::PERMIT_SHIFT),
+                waiters: Mutex::const_new(Waitlist {
+                    queue: LinkedList::new(),
+                    closed: false,
+                }),
+                resource_span: tracing::Span::none(),
+            };
+        }
+
+        cfg_not_trace! {
+            return Self {
+                permits: AtomicUsize::new(permits << Self::PERMIT_SHIFT),
+                waiters: Mutex::const_new(Waitlist {
+                    queue: LinkedList::new(),
+                    closed: false,
+                }),
+            };
         }
```

Comment:
Surely there must be a better way...

Reply:
I would like to believe so. Any ideas?

Reply:
What happens if you do this?
```Rust
return Self {
    permits: AtomicUsize::new(permits << Self::PERMIT_SHIFT),
    waiters: Mutex::const_new(Waitlist {
        queue: LinkedList::new(),
        closed: false,
    }),
    #[cfg(tracing)]
    resource_span: tracing::Span::none(),
};
```

Reply:
Thanks a lot, I have changed it to use this approach in other places as well